### ISSUE TITLE
Adium support

### ISSUE THREAD

**Issue by vit9696** [majn/telegram-purple#496]
Hello,

Following the comments in https://github.com/majn/telegram-purple/issues/479:

Adium upstream is currently hosted on BitBucket, even though it is not obvious to find:
https://bitbucket.org/adium/adium/src/default/

Lurch4Adium project has the necessary scripts (and minor patches) to build Adium on Mac, which I believe can be of good help:
https://github.com/shtrom/Lurch4Adium/tree/lurch-0.6.8+carbons-0.2.1

Can this be reopened and handled accordingly with the information given? Thanks.

**Comment by BenWiederhake** [comment]
Are you willing to do that for us?  @majn's build env is broken, and apparently I'm too stupid to set up a MacOS VM (I gave up after trying for three hours).

Patches and binaries (along with a "Really I promise I didn't put weird extra stuff into those files") would be very welcome.

**Comment by vit9696** [comment]
Not likely in the near future, but I will have a look unless anybody takes it earlier. The description should at least give an idea for other contributors.

**Comment by BenWiederhake** [comment]
Well, we used to support Adium, too, but couldn't build the Adium bundle for the latest release because of the mentioned reasons.  So yeah we kinda do support Adium, you just have to build it yourself.

**Comment by ksuther** [comment]
Attached is a build of telegram-adium 1.4.1. No source changes were required, just some Xcode config updates. Also sending a PR with some updates to the instructions but overall it builds just fine.

[telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/3447605/telegram-adium.AdiumLibpurplePlugin.zip)

**Comment by BenWiederhake** [comment]
Thanks!  Since I don't have a MacOS to try it out, I don't really want to call it "official", but I linked it as "unofficial" on the [release page](https://github.com/majn/telegram-purple/releases/tag/v1.4.1).  Again, thanks!

**Comment by BenWiederhake** [comment]
@ksuther Would you be interested in maintaining the AdiumLibpurple build?  I'd love to retroactively add them to 1.4.1 and 1.4.2.  (I should have asked you earlier; sorry.)

**Comment by ksuther** [comment]
Happy to make builds for 1.4.1 and 1.4.2, I'll try to find some time to do that soon.

**Comment by ksuther** [comment]
telegram-adium 1.4.2: [telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/3613868/telegram-adium.AdiumLibpurplePlugin.zip)
telegram-adium 1.4.2 dSYM (debug symbols): [telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/3613869/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

**Comment by BenWiederhake** [comment]
Thanks!  [Added it.](https://github.com/majn/telegram-purple/releases/tag/v1.4.2)

**Comment by RobFog** [comment]
@ksuther After installing the plugin in Adium 1.5.10.2, Telegram does not show up as an available service:
![Bildschirmfoto 2019-10-16 um 21 13 29](https://user-images.githubusercontent.com/2267373/66955234-d8b4e480-f059-11e9-9548-50830b7c332c.png)
Am I missing something?

**Comment by ksuther** [comment]
The build settings allow running on 10.11 although I haven't tested it. Can you make sure the bundle isn't quarantined (`xattr -l <path to plugin>`)? Also try updating to Adium 1.5.10.4.

**Comment by RobFog** [comment]
Thanks, I tried both. I found this output in the Console:

> 16.10.19 21:24:20,115 Adium[23765]: Error loading /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium:  dlopen(/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium, 265): Library not loaded: @executable_path/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple
  Referenced from: /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium
  Reason: image not found
16.10.19 21:24:20,115 Adium[23765]: Failed to obtain principal class from plugin "telegram-adium.AdiumLibpurplePlugin" ("/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin")! infoDictionary: {
    AIMinimumAdiumVersionRequirement = "1.5.10";
    BuildMachineOSBuild = 18G95;
    CFBundleDevelopmentRegion = English;
    CFBundleExecutable = "telegram-adium";
    CFBundleIdentifier = "org.telegram-purple.telegram-adium";
    CFBundleInfoDictionaryVersion = "6.0";
    CFBundleInfoPlistURL = "Contents/Info.plist -- file://localhost/Users/robertvehse/Library/Application%20Support/Adium%202.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/";
    CFBundleName = "telegram-adium";
    CFBundleNumericVersion = 21135360;
    CFBundlePackageType = BNDL;
    CFBundleShortVersionString = "1.4.2";
    CFBundleSignature = AdIM;
    CFBundleSupportedPlatforms =     (
        MacOSX
    );
    CFBundleVersion = "1.4.2";
    CFPlugInDynamicRegisterFunction = "";
    CFPlugInDynamicRegistration = NO;
    CFPlugInFactories =     {
        "00000000-0000-0000-0000-000000000000" = MyFactoryFunction;
    };
    CFPlugInTypes =     {
        "00000000-0000-0000-0000-000000000000" =         (
            "00000000-0000-0000-0000-000000000000"
        );
    };
    CFPlugInUnloadFunction = "";
    DTCompiler = "com.apple.compilers.llvm.clang.1_0";
    DTPlatformBuild = 10G8;
    DTPlatformVersion = GM;
    DTSDKBuild = 18G74;
    DTSDKName = "macosx10.14";
    DTXcode = 1030;
    DTXcodeBuild = 10G8;
    NSHumanReadableCopyright = "Copyright \U00a9 2014-2017 Matthias Jentsch.";
    NSPrincipalClass = TelegramPlugin;
}
16.10.19 21:24:20,115 Adium[23765]: Failed to initialize Plugin "telegram-adium.AdiumLibpurplePlugin" ("/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin")!

**Comment by ksuther** [comment]
What is the version of libpurple in Adium's bundle (`/Applications/Adium.app/Contents/Frameworks/libpurple.framework/Versions`)?

**Comment by RobFog** [comment]
2.12.0r11b8084bcff4

**Comment by ksuther** [comment]
It's failing to load the library but it seems like it's in the correct place. Does `ls -l /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple` show the correct file?

This build is running on 10.12 and 10.14, but I don't see why it wouldn't work on 10.11.

**Comment by RobFog** [comment]
That gives:
> ls -l /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple
ls: /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple: No such file or directory

Are you sure the "MacOS" belongs there? This is what I am seeing:
![Bildschirmfoto 2019-10-16 um 21 50 37](https://user-images.githubusercontent.com/2267373/66957918-081a2000-f05f-11e9-9ee4-b57c24721d1f.png)

**Comment by ksuther** [comment]
Yeah, MacOS should be there. This is the path it is looking for: `@executable_path/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple`. `@executable_path` is the path to Adium's main binary, which is location at `Adium.app/Contents/MacOS/Adium`.

So this means it looks for libpurple at this combined path (take the path to the Adium binary, remove Adium, then append the library loader path): `/path/to/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple`

What do you get for `ls -l /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions`?

This is all assuming you have Adium in /Applications.

**Comment by RobFog** [comment]
I'm sorry, I was stupid. Ever since you suggested trying 1.5.10.4, I was in fact running a copy of Adium not located in /Applications.

If I run the first Terminal command with 1.5.10.2 which has libpurple version 2.10.12r8d2874a79747 that seems to work:

> RobbiesAlteGurke:~ robertvehse$ ls -l /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.10.12r8d2874a79747/libpurple
-rwxr-xr-x@ 1 robertvehse  admin  5363936  9 Mär  2016 /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.10.12r8d2874a79747/libpurple

The second command gives:

> RobbiesAlteGurke:~ robertvehse$ ls -l /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions
total 8
drwxr-xr-x@ 5 robertvehse  admin  170  9 Mär  2016 2.10.12r8d2874a79747
lrwxr-xr-x  1 robertvehse  admin   20  9 Mär  2016 Current -> 2.10.12r8d2874a79747
RobbiesAlteGurke:~ robertvehse$

**Comment by ksuther** [comment]
Can you make sure you're running 1.5.10.4 and try it again? To be safe try removing any old versions of Adium.

**Comment by RobFog** [comment]
Okay, having moved 1.5.10.4 to the application folder this is what I get:

> RobbiesAlteGurke:~ robertvehse$ ls -l /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple
-rwxr-xr-x@ 1 robertvehse  staff  2225680 27 Apr  2017 /Applications/Adium.app/Contents/MacOS/../Frameworks/libpurple.framework/Versions/2.12.0r11b8084bcff4/libpurple

Console output:

> 16.10.19 22:10:17,616 Adium[24030]: Error loading /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium:  dlopen(/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium, 265): Library not loaded: /usr/local/opt/libgpg-error/lib/libgpg-error.0.dylib
  Referenced from: /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/Resources/libgcrypt.20.dylib
  Reason: image not found
16.10.19 22:10:17,617 Adium[24030]: Failed to obtain principal class from plugin "telegram-adium.AdiumLibpurplePlugin" ("/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin")! infoDictionary: {
    AIMinimumAdiumVersionRequirement = "1.5.10";
    BuildMachineOSBuild = 18G95;
    CFBundleDevelopmentRegion = English;
    CFBundleExecutable = "telegram-adium";
    CFBundleIdentifier = "org.telegram-purple.telegram-adium";
    CFBundleInfoDictionaryVersion = "6.0";
    CFBundleInfoPlistURL = "Contents/Info.plist -- file:///Users/robertvehse/Library/Application%20Support/Adium%202.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/";
    CFBundleName = "telegram-adium";
    CFBundleNumericVersion = 21135360;
    CFBundlePackageType = BNDL;
    CFBundleShortVersionString = "1.4.2";
    CFBundleSignature = AdIM;
    CFBundleSupportedPlatforms =     (
        MacOSX
    );
    CFBundleVersion = "1.4.2";
    CFPlugInDynamicRegisterFunction = "";
    CFPlugInDynamicRegistration = NO;
    CFPlugInFactories =     {
        "00000000-0000-0000-0000-000000000000" = MyFactoryFunction;
    };
    CFPlugInTypes =     {
        "00000000-0000-0000-0000-000000000000" =         (
            "00000000-0000-0000-0000-000000000000"
        );
    };
    CFPlugInUnloadFunction = "";
    DTCompiler = "com.apple.compilers.llvm.clang.1_0";
    DTPlatformBuild = 10G8;
    DTPlatformVersion = GM;
    DTSDKBuild = 18G74;
    DTSDKName = "macosx10.14";
    DTXcode = 1030;
    DTXcodeBuild = 10G8;
    NSHumanReadableCopyright = "Copyright \U00a9 2014-2017 Matthias Jentsch.";
    NSPrincipalClass = TelegramPlugin;
}
16.10.19 22:10:17,617 Adium[24030]: Failed to initialize Plugin "telegram-adium.AdiumLibpurplePlugin" ("/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin")!
16.10.19 22:10:28,016 Adium[24030]: [Accounts] Failed to update account with identifier 981A6D31-9640-4618-815D-6DA7393CF554, error: Error Domain=ABAddressBookErrorDomain Code=1002 "(null)"
16.10.19 22:10:28,454 Adium[24030]: [Accounts] Failed to update account with identifier 9392EC6D-D847-4893-892A-1B0409AF28B4, error: Error Domain=ABAddressBookErrorDomain Code=1002 "(null)"

I did try an older version of the plugin a couple of years ago (the version is the Plugin folder is current one!). Could the line "Failed to update account with identifier " refer to preference cruft from a deleted Telegram account in Adium?

**Comment by ksuther** [comment]
Progress! I see an actual problem now :) It's looking for one of the libraries in the wrong place. Can you try this command?

`install_name_tool -change "/usr/local/opt/libgpg-error/lib/libgpg-error.0.dylib" "@loader_path/libgpg-error.0.dylib" //Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/Resources/libgcrypt.20.dylib`

**Comment by RobFog** [comment]
It errors:

> RobbiesAlteGurke:~ robertvehse$ install_name_tool -change "/usr/local/opt/libgpg-error/lib/libgpg-error.0.dylib" "@loader_path/libgpg-error.0.dylib" //Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/Resources/libgcrypt.20.dylib
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: more than one input file specified (Support/Adium and //Users/robertvehse/Library/Application)
Usage: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool [-change old new] ... [-rpath old new] ... [-add_rpath new] ... [-delete_rpath old] ... [-id name] input
RobbiesAlteGurke:~ robertvehse$

**Comment by ksuther** [comment]
Sorry, I had a bad paste:

`install_name_tool -change "/usr/local/opt/libgpg-error/lib/libgpg-error.0.dylib" "@loader_path/libgpg-error.0.dylib" "/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/Resources/libgcrypt.20.dylib"`

**Comment by RobFog** [comment]
Shucks:

> RobbiesAlteGurke:~ robertvehse$ install_name_tool -change "/usr/local/opt/libgpg-error/lib/libgpg-error.0.dylib" "@loader_path/libgpg-error.0.dylib" "/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/Resources/libgcrypt.20.dylib"
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: object: /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/Resources/libgcrypt.20.dylib malformed object (unknown load command 8)

**Comment by ksuther** [comment]
I see now, that library is built for 10.14 and later. Before it worked on 10.12 before the person using it had libgpg-error installed separately through Homebrew. I'll need to investigate and see what options there are here.

**Comment by RobFog** [comment]
Again, thanks ever so much for your patience!

**Comment by RobFog** [comment]
We are trying to get Adium dev moving again. I am investigating what (protocol) plugins are out there that we can advertise. 🙂

I haven't been using Adium myself much in the recent years but being able to use Telegram in it would be a good reason to get into the habit again.

**Comment by BenWiederhake** [comment]
I love seeing momentum regarding Adium again!  Let me know if I can help with the general build system in any way.  (tgl is undebuggable, but should work for another few years at least; unless #517 breaks our necks.)

**Comment by RobFog** [comment]
You know, @BenWiederhake, I think we might actually call on your services soon. It would be great if we could make building Adium easier – both for the development of Adium itself and for people looking to create/port protocol plugins. 🙂

**Comment by ksuther** [comment]
@RobFog Here is an updated build with the minimum system version set to 10.11 rather than 10.14. Can you give it a try?

[telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/3747139/telegram-adium.AdiumLibpurplePlugin.zip)
[telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/3747138/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

**Comment by RobFog** [comment]
Thanks, @ksuther. 🙂

It doesn't work but there's progress. It seems you fixed the library that caused the error above. Console now reports an issue with a different library:

> 19.10.19 21:53:17,478 Adium[2800]: Error loading /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium:  dlopen(/Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium, 265): Library not loaded: @loader_path/../Frameworks/libtgl.so
  Referenced from: /Users/robertvehse/Library/Application Support/Adium 2.0/PlugIns/telegram-adium.AdiumLibpurplePlugin/Contents/MacOS/telegram-adium
  Reason: image not found

**Comment by ksuther** [comment]
Sorry, let's try this again! If there are more releases then there should probably be a proper automated build process :)

[telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/3747256/telegram-adium.AdiumLibpurplePlugin.zip)
[telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/3747255/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

**Comment by RobFog** [comment]
W00t! Telegram install succeeded:

![Bildschirmfoto 2019-10-19 um 23 51 09](https://user-images.githubusercontent.com/2267373/67152322-62b7b400-f2cb-11e9-9936-c6d891c9b672.png)

I've entered the phone number I use for Telegram. When trying to connect, Adium becomes unresponsive. It recovers briefly, before it tries to connect again. Is a sample (https://jmp.sh/wtwiG7J) or a spindump (https://jmp.sh/kf4wKZO) helpful?

**Comment by BenWiederhake** [comment]
The spindump is helpful, because it tells me that you didn't give it enough time :P 
The "gcry_…" functions you see are busy with the Proof-of-Work part of login (specifically, factorization).  Sadly, tgl runs them on the main thread, so the entire program halts while it's running.

Temporary workaround: Just give it enough time.  Even if it times out, pure chance should ensure that eventually, the PoW code is fast enough.  I hope.

An actual fix would involve getting the PoW to run off-thread, which involves making callback-hell even worse, so I'm not even going to try.

**Comment by RobFog** [comment]
Heh. I tried and now it did work.

A huge thank-you to both of you! 🙃

**Comment by RobFog** [comment]
Would it make sense to replace the download at https://github.com/majn/telegram-purple/releases/tag/v1.4.2 by the bundle @ksuther posted above?

**Comment by BenWiederhake** [comment]
> there should probably be a proper automated build process [for Adium bundles]

Good idea!  A nice first step would be to write down the commands you actually ran.  That's how I started with `mkwindows.sh`, and by now it's fully automatic.

> Would it make sense to replace the download …

@ksuther I don't know anything about MacOS.  Do I need to provide both files, or only one?  If both: Do *all* MacOS users already know which file they need?

**Comment by ksuther** [comment]
The second file contains the debugging symbols if anyone should need them, they're not needed just to run.

The real prerequisite is Adium that builds on a modern toolchain so that telegram-purple can bootstrap off of it. It's probably not too far off, but I didn't spend the time to hunt around fixing the build errors (looks like someone else is trying to do so in the linked forum thread). I'll start to cobble together a shell script for the future as a start though.

**Comment by RobFog** [comment]
> The real prerequisite is Adium that builds on a modern toolchain so that telegram-purple can bootstrap off of it. It's probably not too far off, but I didn't spend the time to hunt around fixing the build errors (looks like someone else is trying to do so in the linked forum thread).

Indeed! We now have two pull requests 🙂 (https://bitbucket.org/adium/adium/pull-requests/11, https://bitbucket.org/adium/adium/pull-requests/12); PR 11 addresses a major build issue. An update to the project file will follow [soon](https://forums.cocoaforge.com/viewtopic.php?f=13&p=140687&sid=5df9ed75ff3a6cfbc9975aeb2cb18173#p140686), hopefully.

>I'll start to cobble together a shell script for the future as a start though.

👍

**Comment by ksuther** [comment]
I added a script to build some of the dependencies to #521.

If Adium becomes buildable on Xcode 11 we could add Adium as a submodule to this project and have it automatically build the proper framework dependencies and dramatically simplify the build instructions. It would be fairly simple to have Travis do the builds after that. This also assumes that the Adium GitHub mirror gets updated.

**Comment by RobFog** [comment]
I will work torwards getting a) Adium buildable on Xcode 11 and b) the GitHub mirror updated.

**Comment by BenWiederhake** [comment]
> Sorry, let's try this again! If there are more releases then there should probably be a proper automated build process :)
> 
> [telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/3747256/telegram-adium.AdiumLibpurplePlugin.zip)
> [telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/3747255/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

I hate to cause trouble, but the built file contains the string `commit: c0187395cd`, but [that commit](https://github.com/majn/telegram-purple/commit/c0187395cd) doesn't contain the changes you mentioned.  Is it inherent to the build process (then I'll just keep it as-is), or did you change any of the files?  If so, I'd like to ask you to commit the changes somewhere (can be a leaf branch), recompile, and push.  I strongly prefer if the binaries out there actually reference the version from which they were built. :)

**Comment by BenWiederhake** [comment]
Oh!  I didn't see #520 yet.  Sorry for my prodding, take your time, and thank you! :D

**Comment by ksuther** [comment]
I didn't realize the commit was in the build. I pulled the changes but didn't run `make commit.h` to update it. That build is for 1.4.2 but I can see why that'd be confusing. I can rebuild with an updated `commit.h` if you'd prefer.

**Comment by BenWiederhake** [comment]
Huh, `commit.h` should have been built automatically by the rest of `make`.  Or does xcode not read the makefile at all?

**Comment by ksuther** [comment]
Xcode is completely independent from make. Now that I know it exists I can have Xcode call `make commit.h` as a build step.

**Comment by RobFog** [comment]
Hi folks, I've got good news:
- I've set up a new Github mirror here: https://github.com/adium/Adium-2 We might end up moving to Github altogether when Bitbucket's support for Mercurial runs out in the summer but this should work for the moment.
- I've merged https://bitbucket.org/adium/adium/pull-requests/11 I need to inquire whether that makes Adium buildable on Xcode 11. Maybe give it a try?

**Comment by ksuther** [comment]
@RobFog It doesn't build from a clean checkout for me. I had to upgrade an old nib `ContactListChangeUserPictureMenu.xib` and then ran into framework signing issues. My initial guess is something about the configurations for the framework targets isn't playing nicely with Xcode's attempts to sign them automatically, but I didn't dig in very far.

**Comment by RobFog** [comment]
> @RobFog It doesn't build from a clean checkout for me. I had to upgrade an old nib `ContactListChangeUserPictureMenu.xib` and then ran into framework signing issues. My initial guess is something about the configurations for the framework targets isn't playing nicely with Xcode's attempts to sign them automatically, but I didn't dig in very far.

When did you last update? In the last three days, Evan, Adium's former lead dev, has been making changes. One of those changes, is supposed to fix the .xib you mention: https://hg.adium.im/adium/rev/78f15a50a998 He has also merged the 1.5.10.x changes into the 1.5.11 branch. The 1.5.11 branch was originally created to add support for Google Talk OAuth 2 support (https://beta.adium.im) but that implementation was never finished. I assume he intends to sort that out.

So perhaps it's best for us here to wait till the release happens.

**Comment by ksuther** [comment]
telegram-adium 1.4.3: [telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/4384129/telegram-adium.AdiumLibpurplePlugin.zip)
telegram-adium 1.4.3 dSYM (debug symbols): [telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/4384124/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

**Comment by RobFog** [comment]
Thanks, @ksuther.

While a new Adium release is still pending (duh!), Asher (https://github.com/Asher-) has made considerable progress in the repo on Github: https://github.com/adium Adium should now run well on current versions of macOS – and work well with current Xcode if I'm not mistaken.

We are currently investigating issues with macOS 10.11 which is the new minimum requirement. We welcome any hints as to what could cause the issues and any suggestions on how to investigate them without direct access to a machine with 10.11.

**Comment by rodneyrod** [comment]
@RobFog Do you happen to have any new builds of Adium to test?

**Comment by RobFog** [comment]
@rodneyrod Hi Rodney. Thanks for asking. Not yet, but soon, hopefully.

**Comment by e1z0** [comment]
Hey,
It would be nice to see a compiled version of plugin for MacOS 32bit, snow leopard lacks of telegram support for some old imac models :'(

Thanks!

**Comment by ksuther** [comment]
telegram-adium 1.4.5: [telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/5743929/telegram-adium.AdiumLibpurplePlugin.zip)
telegram-adium 1.4.5 dSYM (debug symbols): [telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/5743931/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

**Comment by BenWiederhake** [comment]
@ksuther Hmm, it's a modified version 1.4.5, so I don't really like calling it 1.4.5. Could you rebuild it for [1.4.6](https://github.com/majn/telegram-purple/releases/tag/v1.4.6)? Sorry for the extra effort, but let's try to avoid version confusion :)

**Comment by ksuther** [comment]
Sure, here's 1.4.6:

telegram-adium 1.4.6: [telegram-adium.AdiumLibpurplePlugin.zip](https://github.com/majn/telegram-purple/files/5748501/telegram-adium.AdiumLibpurplePlugin.zip)
telegram-adium 1.4.6 dSYM (debug symbols): [telegram-adium.AdiumLibpurplePlugin.dSYM.zip](https://github.com/majn/telegram-purple/files/5748502/telegram-adium.AdiumLibpurplePlugin.dSYM.zip)

**Comment by BenWiederhake** [comment]
[Thanks!](https://github.com/majn/telegram-purple/releases/tag/v1.4.6)
